reference/src/taimpl/src/internal/ta.c

```diff
@@ -371,6 +372,144 @@ static sa_status ta_invoke_key_import(
+	      
+	      /*TODO SoC Vendor:here just provide an example to pass certificate data,
+                how to use it is all up to SOC vendor.
+               */
+	      {
+	          void   *oemDeviceCertificate =
+		  	 ((WidevineOemProvisioning*)params[1].mem_ref)->oemDeviceCertificate;
+	          size_t oemDeviceCertificateLength =
+			 ((WidevineOemProvisioning*)params[1].mem_ref)->oemDeviceCertificateLength;
+		  INFO("oemDeviceCertificate:0x%x, oemDeviceCertificateLength:%d",
+		         oemDeviceCertificate, oemDeviceCertificateLength);
+	      }
+	      break;
+	 case PLAYREADY_MODEL_PROVISIONING:
+	      in = ((PlayReadyProvisioning*)params[1].mem_ref)->privateKey;
+	      in_length = ((PlayReadyProvisioning*)params[1].mem_ref)->privateKeyLength;
+
+	      /*TODO SoC Vendor:here just provide an example to pass certificate data
+                and model type. How to use them is all up to SOC vendor.
+               */
+	      {
+                  void   *modelCertificate =
+		 	 ((PlayReadyProvisioning*)params[1].mem_ref)->modelCertificate;
+	          size_t modelCertificateLength =
+			 ((PlayReadyProvisioning*)params[1].mem_ref)->modelCertificateLength;
+		  INFO("modelCertificate:0x%x, modelCertificateLength:%d",
+		        modelCertificate, modelCertificateLength);
+		  INFO("Model type : %s",
+                  (((PlayReadyProvisioning*)params[1].mem_ref)->modelType == PLAYREADY_MODEL_2K)? 
+		     "model 2000" :
+                  ((((PlayReadyProvisioning*)params[1].mem_ref)->modelType == PLAYREADY_MODEL_3K)?
+		     "model 3000":
+                     "No such model type"));
+	      }
+              break; 
+	 case NETFLIX_PROVISIONING:
+	      in_hmac = ((NetflixProvisioning*)params[1].mem_ref)->hmacKey;
+	      in_hmac_length = ((NetflixProvisioning*)params[1].mem_ref)->hmacKeyLength;
+	      in_wrapping_key = ((NetflixProvisioning*)params[1].mem_ref)->wrappingKey;
+	      in_wrapping_key_length = ((NetflixProvisioning*)params[1].mem_ref)->wrappingKeyLength;
+
+	      /*TODO SoC Vendor:here just provide an example to pass ESN data,
+                how to use it is all up to SOC vendor.
+               */
+              {
+	          void   *esn = ((NetflixProvisioning*)params[1].mem_ref)->esnContainer;
+	          size_t esnLength = ((NetflixProvisioning*)params[1].mem_ref)->esnContainerLength;
+		  INFO("esn:0x%x, esnLength:%d", esn, esnLength);
+	      }
+              break;
+	 default:
+	    ERROR("unknown key provisioning type");
+            break;
+     }
+     INFO("ta_key_type: %d", ta_key_type);
+     INFO("in_length: %d", in_length);
+     INFO("key_import->key: 0x%x", key_import->key);
+     sa_status status = SA_STATUS_OK;
+     if (WIDEVINE_OEM_PROVISIONING == ta_key_type ||
+	 PLAYREADY_MODEL_PROVISIONING == ta_key_type) {	
+         status = ta_sa_key_import(&key_import->key, key_import->key_format,
+	    in,in_length,
+            parameters, context->client, uuid);
```

Comment:
I noticed that you are doing error checking on status below, but not here.  was that by design?

Reply:
Similar to other ta_invoke_key_xxxx functions, ta_invoke_key_provision primarily focuses on returning a status code at the end of its execution.  For reference, you can examine how ta_invoke_key_get_public handles its return value.